lib/concentrate/pipeline/stop_time_updates_pipeline_supervisor.ex

```diff
@@ -0,0 +1,16 @@
+defmodule Concentrate.Pipeline.StopTimeUpdatesPipelineSupervisor do
+  @moduledoc """
+  Supervisor for the StopTimeUpdatesPipeline.
+  """
+  use Supervisor
+
+  @spec start_link(keyword()) :: Supervisor.on_start()
+  def start_link(opts) do
+    Supervisor.start_link(__MODULE__, opts)
+  end
+
+  @impl true
+  def init(opts) do
+    Supervisor.init(Concentrate.Pipeline.StopTimeUpdatesPipeline.init(opts), strategy: :rest_for_one)
```

Comment:
This is also something like you could include in `Pipeline`, since you can pass a list of children to `Supervisor.start_link`. The two separate pipelines would be configured like this:
```
{Concentrate.Pipeline, module: Concentrate.Pipeline.StopTimeUpdates, ..other_opts..},
{Concentrate.Pipeline, module: Concentrate.Pipeline.VehiclePositions, ..other opts..}
```
The `init/1` callback for `Pipeline` behaviors is the same, and then you implement `start_link/1` and `child_spec/1` on the `Pipeline` module directly. This is also a case where you'll want an `id` on the child spec: you can probably use the module for the ID.

Reply:
I feel like I might not be fully grasping what you are saying here, perhaps starting with the bit about "you can pass a list of children to Supervisor.start_link". Are you suggesting something counter to the suggestion Sky made [here](https://github.com/mbta/skate/pull/493#discussion_r389722614) of separating out the pipelines from a single supervisor? Or is what you are saying different?

Reply:
I'll write it up more thoroughly tomorrow morning.

Reply:
Thanks! We can also talk through it in person if that's easier.

Reply:
```elixir
defmodule Pipeline do
  @moduledoc """Create a pipeline supervisor for a given module."""
  
  @callback init(Keyword.t) :: [Supervisor.child_spec]

  def start_link(mod, opts) do
    children = mod.init(opts)
    Supervisor.start_link(children, restart: :rest_for_one)
  end

  def child_spec(opts) do
   mod = Keyword.fetch(opts, :module)
    %{
      type: :supervisor, 
      id: mod,
      start: {__MODULE__, :start_link, [mod, opts]}
    }
  end
end

# in Concentrate.Supervisor
def start_link(...) do
  Supervisor.start_link([
    {Pipeline, module: StopTimePipeline, ...},
    {Pipeline, module: VehiclePositionSupervisor, ...}
  ], restart: :one_for_one)
end

Reply:
Thanks, that's helpful. Let me know if you have any suggestions on my adaptation.